compiler/main/checks.cpp

```diff
@@ -966,6 +966,15 @@ checkFormalActualTypesMatch()
           continue;
         }
 
+        if ((isCPtrConstChar(formal->getValType()) ||
+             isCPtrConstChar(actual->getValType())) &&
+            (formal->getValType()==dtStringC ||
+             actual->getValType()==dtStringC)) {
+            // we allow conversion between these types in function resolution
+            // TODO: remove this once we get rid of c_string remnants
+            continue;
+        }
+
```

Comment:
This ordering of conditionals threw me off for a sec (I read it as "both formal and actual are c strings or both formal and actual are strings" and was confused why plain equality wouldn't work). I think I'd prefer it in the (string , cstring) or (cstring, string) form. But that's just a minor nit and the PR is good as is.

Reply:
I swear I got that pattern from somewhere in the existing code, and it threw me first too, but then I made sense of it and I kind of liked it, which is why I repeated it here 😃